src/codeu/chat/server/Controller.java

```diff
@@ -56,6 +59,326 @@ public ConversationHeader newConversation(String title, Uuid owner) {
   }
 
   @Override
+  public Integer toggleRemovedBit(Uuid convo, Uuid user){
+    final ConversationHeader foundConvo = model.conversationById().first(convo);
+    final User foundUser = model.userById().first(user);
+
+    Integer result = 0;
+
+    if(foundConvo != null && foundUser != null) {
+      Integer access = foundConvo.accessControls.computeIfAbsent(user, newAccess -> 0);
+      Integer newAccess;
+
+      // Flag is set to true if the user is removed from a conversation.
+      // Flag stays true once it's set.
+      newAccess = access | ConversationHeader.REMOVED;
+
+      result = newAccess;
+      foundConvo.accessControls.put(user, newAccess);
+
+      LOG.info(
+              "toggleRemovedBit success (user.id=%s conversation.id=%s access=%s)",
+              foundUser.id,
+              foundConvo.id,
+              newAccess);
+    } else {
+      LOG.info(
+              "toggleRemovedBit fail (user.id=%s conversation.id=%s)",
+              foundUser.id,
+              foundConvo.id);
+    }
+
+    return result;
+  }
+
+  @Override
+  public Integer toggleCreatorBit(Uuid convo, Uuid user, Boolean flag){
+    final ConversationHeader foundConvo = model.conversationById().first(convo);
+    final User foundUser = model.userById().first(user);
+
+    Integer result = 0;
+
+    if(foundConvo != null && foundUser != null) {
+      Integer access = foundConvo.accessControls.computeIfAbsent(user, newAccess -> 0);
+      Integer newAccess;
+
+      // If user is to be set as a Creator, it will also be it's children controls
+      if(flag) {
+        newAccess = access | ConversationHeader.CREATOR;
+        newAccess |= ConversationHeader.OWNER;
+        newAccess |= ConversationHeader.MEMBER;
+      }
+      else
+        newAccess = access & ~ConversationHeader.CREATOR;
+
+      result = newAccess;
+      foundConvo.accessControls.put(user, newAccess);
+
+      LOG.info(
+              "toggleCreatorBit success (user.id=%s conversation.id=%s access=%s)",
+              foundUser.id,
+              foundConvo.id,
+              newAccess);
+    } else {
+      LOG.info(
+              "toggleCreatorBit fail (user.id=%s conversation.id=%s)",
+              foundUser.id,
+              foundConvo.id);
+    }
+
+    return result;
+  }
+
+  @Override
+  public Integer toggleOwnerBit(Uuid convo, Uuid user, Boolean flag){
```

Comment:
Hmm..I looked over the code to see to implement this, and I think that the `toggleAccessBit` method would just get too long? I also have some specific log statements in each specific bit toggle method currently.